cookbooks/aws-parallelcluster-config/templates/default/directory_service/update_directory_service_password.sh.erb

```diff
@@ -20,20 +20,34 @@ REGION="<%= @region %>"
 
 PYTHON_CODE_READ_CONFIG="import configparser;file='${SSSD_CONFIG_FILE}';config=configparser.ConfigParser();config.read(file)"
 
-echo "Reading password from ${SSSD_CONFIG_FILE}"
+echo "[INFO] Reading password from ${SSSD_CONFIG_FILE}"
 password_from_sssd_config=$(python3 -c "${PYTHON_CODE_READ_CONFIG}; print(config['${SSSD_SECTION}']['${SSSD_PROPERTY}'])")
 
-echo "Reading password from AWS Secrets Manager: ${SECRET_ARN}"
-password_from_secrets_manager=$(aws secretsmanager get-secret-value --secret-id ${SECRET_ARN} --region ${REGION} --query 'SecretString' --output text)
-
-[ "${password_from_sssd_config}" == "${password_from_secrets_manager}" ] && echo "Password match, skipping update" && exit 0
-
-echo "Writing AWS Secrets Manager password to ${SSSD_CONFIG_FILE}"
-python3 -c "${PYTHON_CODE_READ_CONFIG}; config['${SSSD_SECTION}']['${SSSD_PROPERTY}']='${password_from_secrets_manager}'; config.write(open(file,'w'))"
-echo "Password updated in ${SSSD_CONFIG_FILE}"
-cp ${SSSD_CONFIG_FILE} ${SSSD_SHARED_CONFIG_FILE}
-echo "${SSSD_CONFIG_FILE} copied to ${SSSD_SHARED_CONFIG_FILE}"
-
-echo "Restarting services"
+echo "[INFO] Reading password from AWS: ${SECRET_ARN}"
+service=$(echo "${SECRET_ARN}" | cut -d ':' -f 3)
+resource=$(echo "${SECRET_ARN}" | cut -d ':' -f 6-)
```

Comment:
same here it would be useful to see fthe format of the parameter we're parsing to better understand the code in the two cases.

Reply:
Question: what does the `6-` mean?

Reply:
Examples of ARNs will be added in the next revision.
`6-` means that fields from the 6th onwards will be returned. This is required because an ARN is made of `arn:PARTITION:SERVICE:REGION:ACCOUNT:RESOURCE`
1. for a secret in Secrets Manager, the RESOURCE field must be `secret:PasswordName`
2. for a parameter in SSM, the RESOURCE field must be `parameter/PasswordName`